pkg/gatherers/common/gather_logs.go

```diff
@@ -156,38 +178,35 @@ func filterLogs(
 	}()
 
 	scanner := bufio.NewScanner(stream)
-	return FilterLogFromScanner(scanner, messagesToSearch, regexSearch, nil)
+	return FilterLogFromScanner(scanner, messagesToSearch, messagesRegexp, nil)
 }
 
 // FilterLogFromScanner filters the desired messages from the log
-func FilterLogFromScanner(scanner *bufio.Scanner, messagesToSearch []string, regexSearch bool,
+func FilterLogFromScanner(scanner *bufio.Scanner, messagesToSearch []string, messagesRegexp *regexp.Regexp,
 	cb func(lines []string) []string) (string, error) {
 	var result []string
 
-	var messagesRegexp *regexp.Regexp
-	if regexSearch {
-		messagesRegexp = regexp.MustCompile(strings.Join(messagesToSearch, "|"))
-	}
-
 	for scanner.Scan() {
 		line := scanner.Text()
-		if len(messagesToSearch) == 0 {
-			result = append(result, line)
-			continue
-		}
 
-		if regexSearch && messagesRegexp != nil {
+		if messagesRegexp != nil {
 			matches := messagesRegexp.MatchString(line)
 			if matches {
 				result = append(result, line)
 			}
-		} else {
-			for _, messageToSearch := range messagesToSearch {
-				if strings.Contains(strings.ToLower(line), strings.ToLower(messageToSearch)) {
-					result = append(result, line)
-				}
+			continue
+		}
+
+		for _, messageToSearch := range messagesToSearch {
+			if strings.Contains(strings.ToLower(line), strings.ToLower(messageToSearch)) {
+				result = append(result, line)
 			}
 		}
+
+		if len(messagesToSearch) == 0 {
+			result = append(result, line)
+			continue
+		}
```

Comment:
I don't understand the need of this 4 lines of code. Can you elaborate please?

Reply:
I'll remove this block... We never collect any log without filtering it and we probably never want to do it.

Reply:
Oh, actually we need this block, but I kind of changed it a little bit. There is one gather that kind of doesn't filter the log, actually, it gets just the last 50 lines.